### PR TITLE
Feature/aurora postgres plugin

### PR DESCRIPTION
**Pull request by ibarbashov**

### CODE REVIEW

File: aurora-postgresql-plugin/src/main/java/co/cask/auroradb/postgres/AuroraPostgresAction.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+  /**
+   * Aurora DB PostgreSQL Action Config.
+   */
+  public static class AuroraPostgresActionConfig extends DBSpecificQueryConfig {
+
+    @Name(AuroraPostgresConstants.CONNECTION_TIMEOUT)
+    @Description("The timeout value used for socket connect operations. If connecting to the server takes longer" +
+      " than this value, the connection is broken. " +
+      "The timeout is specified in seconds and a value of zero means that it is disabled")
```
Comment (jayfjin) [inline aurora-postgresql-plugin/src/main/java/co/cask/auroradb/postgres/AuroraPostgresAction.java]: nit: period at end of sentence

File: aurora-postgresql-plugin/src/main/java/co/cask/auroradb/postgres/AuroraPostgresPostAction.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    this.auroraPostgresQueryActionConfig = auroraPostgresQueryActionConfig;
+  }
+
+  /**
+   * Aurora DB PostgreSQL post action configuration.
+   */
+  public static class AuroraPostgresQueryActionConfig extends DBSpecificQueryActionConfig {
+
+    @Name(AuroraPostgresConstants.CONNECTION_TIMEOUT)
+    @Description("The timeout value used for socket connect operations. If connecting to the server takes longer" +
```
Comment (jayfjin) [inline aurora-postgresql-plugin/src/main/java/co/cask/auroradb/postgres/AuroraPostgresPostAction.java]: Can this description be extracted to the constants class? I see it in two different places

File: aurora-postgresql-plugin/src/main/java/co/cask/auroradb/postgres/AuroraPostgresSink.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    super.dbColumns = columnsJoiner.toString();
+  }
+
+  /**
+   * Aurora DB PostgreSQL action configuration.
+   */
+  public static class AuroraPostgresSinkConfig extends DBSpecificSinkConfig {
+
+    @Name(AuroraPostgresConstants.CONNECTION_TIMEOUT)
+    @Description("The timeout value used for socket connect operations. If connecting to the server takes longer" +
```
Comment (jayfjin) [inline aurora-postgresql-plugin/src/main/java/co/cask/auroradb/postgres/AuroraPostgresSink.java]: same here

File: aurora-postgresql-plugin/src/main/java/co/cask/auroradb/postgres/AuroraPostgresSink.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+/**
+ * Sink support for an Aurora DB PostgreSQL database.
+ */
+@Plugin(type = BatchSink.PLUGIN_TYPE)
+@Name(AuroraPostgresConstants.PLUGIN_NAME)
+@Description("Writes records to a table of Aurora DB PostgreSQL cluster. " +
+  "Each record will be written in a row in the table.")
+public class AuroraPostgresSink extends AbstractDBSink {
+  private static final Character ESCAPE_CHAR = '"';
```
Comment (jayfjin) [inline aurora-postgresql-plugin/src/main/java/co/cask/auroradb/postgres/AuroraPostgresSink.java:43]: does escape char need to be \\" or is " fine?

File: aurora-postgresql-plugin/src/main/java/co/cask/auroradb/postgres/AuroraPostgresSource.java [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    return AuroraPostgresDBRecord.class;
+  }
+
+  /**
+   * Aurora DB PostgreSQL source config.
+   */
+  public static class AuroraPostgresSourceConfig extends DBSpecificSourceConfig {
+
+    @Name(AuroraPostgresConstants.CONNECTION_TIMEOUT)
+    @Description("The timeout value used for socket connect operations. If connecting to the server takes longer" +
```
Comment (jayfjin) [inline aurora-postgresql-plugin/src/main/java/co/cask/auroradb/postgres/AuroraPostgresSource.java]: here too

File: database-commons/src/main/java/co/cask/db/batch/sink/AbstractDBSink.java [hunk 6]
```diff
@@ -336,6 +338,16 @@ private void emitLineage(BatchSinkContext context, List<Schema.Field> fields) {
     @Description("Name of the database table to write to.")
     @Macro
     public String tableName;
+
+    /**
+     * Adds escape characters (back quotes, double quotes, etc.) to the table name for
```
Comment (jayfjin) [inline database-commons/src/main/java/co/cask/db/batch/sink/AbstractDBSink.java:343]: so the escape character used depends on the underlying DB?

Comment (jayfjin) [reply database-commons/src/main/java/co/cask/db/batch/sink/AbstractDBSink.java:343]: Also why does only the sink need this and not the source?

**Review by jayfjin — COMMENTED** [review]
Just some minor comments/questions

File: aurora-postgresql-plugin/src/main/java/co/cask/auroradb/postgres/AuroraPostgresAction.java (hunk 1 above)
Comment (ibarbashov) [reply aurora-postgresql-plugin/src/main/java/co/cask/auroradb/postgres/AuroraPostgresAction.java]: done

File: aurora-postgresql-plugin/src/main/java/co/cask/auroradb/postgres/AuroraPostgresPostAction.java (hunk 2 above)
Comment (ibarbashov) [reply aurora-postgresql-plugin/src/main/java/co/cask/auroradb/postgres/AuroraPostgresPostAction.java]: done

File: aurora-postgresql-plugin/src/main/java/co/cask/auroradb/postgres/AuroraPostgresSink.java (hunk 3 above)
Comment (ibarbashov) [reply aurora-postgresql-plugin/src/main/java/co/cask/auroradb/postgres/AuroraPostgresSink.java]: done

File: aurora-postgresql-plugin/src/main/java/co/cask/auroradb/postgres/AuroraPostgresSink.java (hunk 4 above)
Comment (ibarbashov) [reply aurora-postgresql-plugin/src/main/java/co/cask/auroradb/postgres/AuroraPostgresSink.java:43]: it's fine

File: aurora-postgresql-plugin/src/main/java/co/cask/auroradb/postgres/AuroraPostgresSource.java (hunk 5 above)
Comment (ibarbashov) [reply aurora-postgresql-plugin/src/main/java/co/cask/auroradb/postgres/AuroraPostgresSource.java]: done

File: database-commons/src/main/java/co/cask/db/batch/sink/AbstractDBSink.java (hunk 6 above)
Comment (ibarbashov) [reply database-commons/src/main/java/co/cask/db/batch/sink/AbstractDBSink.java:343]: escape character in PostgreSQL is impacting only the case sensitivity of column names, that's why we need to handle it in sink, but in source Users provides query with column names and we don't infering the schema

Comment (jayfjin) [reply database-commons/src/main/java/co/cask/db/batch/sink/AbstractDBSink.java:343]: sg

**Pull request merged**